### DISCUSSION TITLE
Kernel Memory without OpenAI

### DISCUSSION THREAD

**Discussion by arafattehsin** [microsoft/semantic-kernel#8359] [Q&A] [answered]
Hi Team

I couldn't find any answer related to this so thought to post it here. I have been able to load the model locally with local embeddings. This means that I am not leveraging internet to do text-based RAG and it is working just fine. 

I have written down [a blog post](https://arafattehsin.com/ai-copilot-offline-phi3-semantic-kernel/) on how I have achieved it using `SemanticTextMemory`

Now I want to add documents to it and I am aware that `SemanticTextMemory` can't handle it. Therefore, I am looking for other options and Kernel Memory is one of the primary candidate in that list. However, after exploring it, I have found out that it is tightly coupled with OpenAI services (for embeddings). Is there a way we can use them without OpenAI service and leverage whatever I am using in my existing application. 

P.S: If you have not read my blog post, I am using `bge-micro-v2` for local embeddings and `ITextEmbeddingGenerationService` for text generation. 

Thanks.

**Comment by sophialagerkranspandey** [comment]
We are wondering if you are using Semantic Memory or kernel memory here (https://github.com/microsoft/kernel-memory)?

**Reply by arafattehsin** [reply, marked as answer]
My apologies @sophialagerkranspandey, I am using Kernel Memory but I just realised that I should have posted in their discussion forum :) I will close this discussion.
